### PR TITLE
ci: added yarpgen workflow

### PR DESCRIPTION
**Pull request by aneshlya**
Workflow to run Yarpgen. 
The timeouts for scheduled runs will be adjusted as soon as we fix current findings. Currently it is 240 minutes.

### CODE REVIEW

File: .github/workflows/yarpgen.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        - name: Set timeout for scheduled run
+          if: github.event_name == 'schedule'
+          run: echo "TIMEOUT=240" >> $GITHUB_ENV
+
+        - name: Run YARPGEN
+          run: |
+            $YARPGEN_HOME/scripts/run_gen.py --std ispc --target ispc --timeout $TIMEOUT
+            tar -czvf yarpgen.tar.gz testing
+
+        - name: Upload YARPGEN results
```
Comment (dbabokin) [inline .github/workflows/yarpgen.yml:108]: Possibly do `ls testing/*/*` to list the seeds is also useful.

**Review by dbabokin — APPROVED** [review]
Looks good!

**Comment by dbabokin** [comment]
Also CC @Vsevolod-Livinskij - just FYI.

File: .github/workflows/yarpgen.yml (hunk 1 above)
Comment (aneshlya) [reply .github/workflows/yarpgen.yml:112]: Added

**Comment by Vsevolod-Livinskij** [comment]
Awesome, I'm glad that you added it to CI! 
Could you please mark detected bugs with a `YARPGen` label (e.g., #1851) or note them somewhere?
This way, we can [keep track of discovered bugs](https://github.com/intel/yarpgen/blob/main/bugs.rst#ispc) and use them in the papers to prove the practical impact of the YARPGen.
Thank you!

**Pull request merged**